example/02-kad/factory.hpp

```diff
@@ -0,0 +1,30 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef LIBP2P_KAD_EXAMPLE_FACTORY_HPP
+#define LIBP2P_KAD_EXAMPLE_FACTORY_HPP
+
+#include <libp2p/crypto/crypto_provider.hpp>
+#include <libp2p/crypto/key_marshaller.hpp>
+#include <libp2p/host/host.hpp>
+#include <memory>
+
+namespace libp2p::kad_example {
+  std::shared_ptr<boost::asio::io_context> createIOContext();
```

Comment:
io_context is created by injector. Why do we need another one? Example: https://github.com/soramitsu/libp2p/blob/master/example/01-echo/libp2p_echo_client.cpp#L37

Reply:
Here is such a thing: io_context is shared among all Host instances in the example, other objects are per-host